projects/ngneat/helipopper/src/lib/helipopper.service.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ApplicationRef,
+  ComponentFactoryResolver,
+  ElementRef,
+  Inject,
+  Injectable,
+  Injector,
+  NgZone,
+  TemplateRef
+} from '@angular/core';
+import { HELIPOPPER_CONFIG, HelipopperConfig, HelipopperOptions } from './helipopper.types';
+import { HelipopperDirective } from './helipopper.directive';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HelipopperService {
+  constructor(
+    private appRef: ApplicationRef,
+    private zone: NgZone,
+    private resolver: ComponentFactoryResolver,
+    private hostInjector: Injector,
+    @Inject(HELIPOPPER_CONFIG) private config: HelipopperConfig
+  ) {}
+
+  open(host: ElementRef, helipopper: string | TemplateRef<any>, options?: HelipopperOptions) {
```

Comment:
It could be also a component and we should allow passing a custom Injector.

Reply:
sure. About the injector part i'll update the actual README as i created the option interface based on it and the @Input injector is missing